jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererUtil.kt

```diff
@@ -36,6 +37,16 @@ import software.aws.toolkits.jetbrains.utils.notifyWarn
 import software.aws.toolkits.resources.message
 import software.aws.toolkits.telemetry.CodewhispererCompletionType
 
+fun checkIfIamIdentityCenterConnection(project: Project, callback: (connection: ToolkitConnection) -> Unit) =
```

Comment:
nit: `runIfIamIdentityCenterConnection` should be a better name for this func. `check` sounds like it will return a boolean

Reply:
It seems like `T` requires a non-Unit to be returned, I will probably change it to something like `Any?`.

Reply:
nope, if the purpose is to have a Unit return type then it's file,  we can just add another one `calculateIf` when we need a return type which consumes a callable instead of runnable.

Reply:
changing back to Unit.